packages/swing-store/src/transcriptStore.js

```diff
@@ -679,20 +722,35 @@ export function makeTranscriptStore(
     }
   }
 
-  function assertComplete(level) {
-    assert.equal(level, 'operational'); // for now
-    // every 'isCurrent' transcript span must have all items
+  function assertComplete(artifactMode) {
+    if (artifactMode === 'debug') {
+      return; // no validation in this mode
+    }
     for (const rec of sqlGetCurrentSpanMetadata.iterate()) {
       const { vatID, startPos, endPos, incarnation } = rec;
-      const count = sqlCountPopulatedSpanItems.get(
-        vatID,
-        incarnation,
-        startPos,
-        endPos,
-      );
+      // at 'operational' and above, every 'isCurrent' transcript span
+      // must have all items
+      const count = sqlCountSpanItems.get(vatID, startPos, endPos);
       if (count !== endPos - startPos) {
         throw Fail`incomplete current transcript span: ${count} items, ${rec}`;
       }
+      if (artifactMode === 'replay' || artifactMode === 'archival') {
+        // at 'replay' and above, every vat's current incarnation must
+        // be fully populated
```

Comment:
The test for current incarnation is strictly a subset of the test for all incarnations, right? Why test "archival" twice?
(arguably same can be said of operational being a subset of both replay and archival)

Reply:
I guess I felt it was more clear what is being asserted, and doesn't require the reader to be as clear about the overlap between the different checks, and it's not like this is a big performance concern. Also it made the error messages more specific: `assertComplete('archival')` on a DB which lacks `operational` would complain about the `incomplete current transcript span` instead of `incomplete archival transcript` (but I guess that's not necessarily a very useful distinction). A non-duplicating approach would look like:
```js
    for (const rec of sqlGetCurrentSpanMetadata.iterate()) {
      const { vatID, startPos, endPos, incarnation } = rec;

      if (artifactMode === 'operational') {
        // at 'operational', every 'isCurrent' transcript span must
        // have all items
        const count = sqlCountSpanItems.get(vatID, startPos, endPos);
        if (count !== endPos - startPos) {
          throw Fail`incomplete current transcript span: ${count} items, ${rec}`;
        }
      } else if (artifactMode === 'replay') {
        // at 'replay', every vat's current incarnation must be fully
        // populated (which implies 'operational')
        const incStart = sqlGetStartOfIncarnation.get(vatID, incarnation);
        const incCount = sqlCountSpanItems.get(vatID, incStart, endPos);
        if (incCount !== endPos - incStart) {
          throw Fail`incomplete current incarnation transcript: ${incCount} items`;
        }
      } else if (artifactMode === 'archival') {
        // at 'archival', every incarnation must be fully populated,
        // which means position=0 up through endPos-1 (which implies
        // 'replay')
        const arcCount = sqlCountSpanItems.get(vatID, 0, endPos);
        if (arcCount !== endPos) {
          throw Fail`incomplete archival transcript: ${arcCount} vs ${endPos}`;
        }
      }
    }
```

I concur that this looks better.. I'll change it, along with the tests that asserted the particular error message being raised.